pkg/webhook/customresourcedefinition/customresourcedefinition_webhook.go

```diff
@@ -0,0 +1,65 @@
+package customresourcedefinition
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"go.goms.io/fleet/pkg/webhook/validation"
+)
+
+const (
+	// ValidationPath is the webhook service path which admission requests are routed to for validating custom resource definition resources.
+	ValidationPath = "/validate-v1-customresourcedefinition"
+	groupMatch     = `^[^.]*\.(.*)`
+)
+
+// Add registers the webhook for K8s bulit-in object types.
+func Add(mgr manager.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &customResourceDefintionValidator{Client: mgr.GetClient()}})
+	return nil
+}
+
+type customResourceDefintionValidator struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func (v *customResourceDefintionValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
A very minor concern here: 

the controller runtime package is now recommending the new CustomValidator/Validator interface; though I really do not have a preference here, the traditional Handle interface would work just fine.